native/core/src/execution/datafusion/expressions/cast.rs

```diff
@@ -1709,6 +1722,37 @@ mod tests {
         assert_eq!(result.len(), 2);
     }
 
+    #[test]
+    fn test_cast_dict_string_to_dict_timestamp() -> DataFusionResult<()> {
```

Comment:
Could you also add a version of this test that casts to a regular timestamp (not dictionary encoded), since that is the primary use case.

Reply:
Hey this is named incorrectly. I have changed it.

This test actually tests 'dict to timstamp array' cast.